### PR TITLE
Add description to assert in `history.dart`

### PR DESCRIPTION
**Pull request by kenzieschmoll**
I have hit this assertion error many times in tests and this message would be helpful for debugging.

### CODE REVIEW

File: lib/web_ui/lib/src/engine/navigation/history.dart [hunk 1]
```diff
@@ -217,7 +217,12 @@ class MultiEntriesBrowserHistory extends BrowserHistory {
     _isTornDown = true;
 
     // Restores the html browser history.
-    assert(_hasSerialCount(currentState));
+    assert(
+      _hasSerialCount(currentState),
+      currentState == null
+          ? 'unexpected null state'
+          : 'state is missing field \'serialCount\'',
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/navigation/history.dart:224]:
```suggestion
          ? 'unexpected null history state'
          : 'history state is missing field \'serialCount\'',
```

**Review by mdebbar — APPROVED** [review]
LGTM, thanks for the improvement!

**Comment by Hixie** [comment]
test-exempt: assert message

...but i wouldn't shy away from testing this. I also wouldn't shy away from making the errors even more useful. You could give context (what is this history state error about?), you could give a solution (what should the developer do when they hit this?). See also https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#error-messages-should-be-useful

**Comment by kenzieschmoll** [comment]
>  You could give context (what is this history state error about?), you could give a solution (what should the developer do when they hit this?)

@mdebbar do you have suggestions on how we can make the error message more actionable? I have only hit this error in flutter web integration test environments, and even for the issue I was hitting, this error message didn't help me solve the problem. My problem was calling
```
html.window.history.replaceState(null, '', newUri.toString());
```
instead of 
```
html.window.history.replaceState(html.window.history.state, '', newUri.toString());
```
But I don't know whether this assert I hit is only applicable to my use case or to many potential causes.

**Pull request merged**